server.js

```diff
@@ -1,54 +1,85 @@
-
-		res.write('Hello OpenShift World! This is a Node.js-based sample application.\n');
-		res.write('Host: ' + req.headers.host + '\n');
-		res.write('\n');
-		res.write('node.js Production Mode: ' + (nodeEnv == 'production' ? 'yes' : 'no') + '\n');
-		res.write('\n');
-		res.write('HTTP/' + req.httpVersion +'\n');
-		res.write('Request headers:\n');
-		res.write(util.inspect(req.headers, null) + '\n');
-		res.write('Request query:\n');
-		res.write(util.inspect(url_parts.query, null) + '\n');
-		res.write('Request body:\n');
-		res.write(util.inspect(formattedBody, null) + '\n');
-		res.write('\n');
-		res.write('Host: ' + os.hostname() + '\n');
-		res.write('OS Type: ' + os.type() + '\n');
-		res.write('OS Platform: ' + os.platform() + '\n');
-		res.write('OS Arch: ' + os.arch() + '\n');
-		res.write('OS Release: ' + os.release() + '\n');
-		res.write('OS Uptime: ' + os.uptime() + '\n');
-		res.write('OS Free memory: ' + os.freemem() / 1024 / 1024 + 'mb\n');
-		res.write('OS Total memory: ' + os.totalmem() / 1024 / 1024 + 'mb\n');
-		res.write('OS CPU count: ' + os.cpus().length + '\n');
-		res.write('OS CPU model: ' + os.cpus()[0].model + '\n');
-		res.write('OS CPU speed: ' + os.cpus()[0].speed + 'mhz\n');
-		res.end('\n');
+var port = process.env.PORT || process.env.OPENSHIFT_NODEJS_PORT || 8080;
+var ip   = process.env.IP   || process.env.OPENSHIFT_NODEJS_IP || '0.0.0.0';
+var mongoURL = process.env.OPENSHIFT_MONGODB_DB_URL || process.env.MONGO_URL;
+if (mongoURL == null && process.env.DATABASE_SERVICE_NAME) {
+	var mongoServiceName = process.env.DATABASE_SERVICE_NAME.toUpperCase();
+	var mongoHost = process.env[mongoServiceName + "_SERVICE_HOST"];
+	var mongoPort = process.env[mongoServiceName + "_SERVICE_PORT"];
+	var mongoUser = process.env.MONGODB_USER
+	if (mongoHost && mongoPort && process.env.MONGODB_DATABASE) {
+      mongoURL = 'mongodb://';
+      if (process.env.MONGODB_USER && process.env.MONGODB_PASSWORD) {
+        mongoURL += process.env.MONGODB_USER + ':' + process.env.MONGODB_PASSWORD + '@';
+	  }
+      mongoURL += mongoHost + ':' + mongoPort + '/' + process.env.MONGODB_DATABASE;
+	}
+}
+var db = null;
+var dbDetails = new Object();
+
+var initDb = function(callback) {
+	if (mongoURL == null) return;
+
+	var mongodb = require('mongodb');	
+	if (mongodb == null) return;
+	
+	mongodb.connect(mongoURL, function(err, conn) {
+		if (err) {
+			callback(err);
+			return;
+		}
+
+		db = conn;
+		dbDetails.databaseName = db.databaseName;
+		dbDetails.url = db.options.url;
+	  	dbDetails.type = 'MongoDB';
+
+		console.log("Connected to MongoDB at: " + mongoURL);
 	});
-	next();
+};
+
+app.get('/', function (req, res) {
+  if (db) {
+	  var col = db.collection('counts');
+	  // Create a document with request IP and current time of request
+	  col.insert({ip: req.ip, date: Date.now()});
+	  col.count(function(err, count){
+ 		res.render('index.html', { pageCountMessage : count, dbInfo: dbDetails });
```

Comment:
I wouldn't send the username and password to the welcome page...
I'm seeing:

```
DB Connection Info:
Type:	MongoDB
URL:	mongodb://userMJF:YGpv6qavy1mPwhYj@172.30.130.208:27017/sampledb
```

I'd just show the server info like `MongoDB (172.30.130.208:27017/sampledb)` or `MongoDB (172.30.130.208:27017)`